src/Visualization/Python/GenerateTetrahedralConnectivity.py

```diff
@@ -0,0 +1,208 @@
+      coordinates: Optional. Name of coordinates dataset. Default:
+        "InertialCoordinates".
+      delete_tetrahedral_connectivity: Optional. Delete the dataset and return.
+        Default: False
+    """
+    h5files = [(h5py.File(filename, "a"), filename) for filename in h5files]
+
+    if not subfile_name:
+        subfiles = available_subfiles(
+            (h5file for h5file, _ in h5files), extension=".vol"
+        )
+        raise RequiredChoiceError(
+            (
+                "Specify '--subfile-name' / '-d' to select a"
+                " subfile containing volume data."
+            ),
+            choices=subfiles,
+        )
+
+    if not subfile_name.endswith(".vol"):
+        subfile_name += ".vol"
+
+    for h5file, filename in h5files:
+        # Open subfile
+        try:
+            vol_subfile = h5file[subfile_name]
+        except KeyError as err:
+            raise ValueError(
+                f"Could not open subfile name '{subfile_name}' in '{filename}'."
+                " Available subfiles: "
+                + str(available_subfiles(h5file, extension=".vol"))
+            ) from err
+
+        # Sort timesteps by time
+        temporal_ids_and_values = sorted(
+            [
+                (key, vol_subfile[key].attrs["observation_value"])
+                for key in vol_subfile.keys()
+            ],
+            key=lambda key_and_time: key_and_time[1],
+        )
+
+        # Stride through timesteps
+        for temporal_id, time in temporal_ids_and_values[::stride]:
+            # Filter by start and end time
+            if start_time is not None and time < start_time:
+                continue
+            if stop_time is not None and time > stop_time:
+                break
+
+            data_at_id = vol_subfile[temporal_id]
+
+            if delete_tetrahedral_connectivity:
+                if "tetrahedral_connectivity" in data_at_id:
+                    del data_at_id["tetrahedral_connectivity"]
+                continue
+
+            x_coords = np.asarray(data_at_id[coordinates + "_x"])
+            y_coords = np.asarray(data_at_id[coordinates + "_y"])
+            if (coordinates + "_z") in data_at_id:
+                z_coords = np.asarray(data_at_id[coordinates + "_z"])
+                coords = np.column_stack((x_coords, y_coords, z_coords))
+            else:
+                coords = np.column_stack((x_coords, y_coords))
+
+            logger.info(
+                f"Generating tetrahedral connectivity at {temporal_id}/{time}. "
+                "This may take a few minutes and may even fail depending on "
+                "the grid structure."
+            )
+            delaunay = spatial.Delaunay(coords)
+            data_at_id.create_dataset(
+                "tetrahedral_connectivity", data=delaunay.simplices.flatten()
+            )
+
+    for h5file in h5files:
+        h5file[0].close()
+
+
+@click.command(
+    name="generate-tetrahedral-connectivity",
```

Comment:
(optional) Maybe just name `triangulate`?

Reply:
I don't feel too strongly, I used this so it matches up with GenerateXdmf `use-tetrahedral-connectivity` but I agree it's very long

Reply:
Up to you